server/jobs/actions.py

```diff
@@ -0,0 +1,84 @@
+import json
+from functools import wraps
+from flask import Blueprint, request, jsonify
+from server.crawler.tracking import job_repository, CrawlJobStatus
+from .utils import token_required
+from bson.errors import BSONError
+
+bp = Blueprint("job_actions", __name__, url_prefix="/api/jobs")
+
+
+def validate_job_id(f):
+    @wraps(f)
+    def decorator(*args, **kwargs):
+        job_id = request.view_args["job_id"]
+        if job_id is None or not task_exist(job_id):
+            return jsonify(
+                message=f"Cannot find job with id {job_id}"
+            ), 404
+        else:
+            return f(*args, **kwargs)
+
+    def task_exist(job_id) -> bool:
+        try:
+            job = job_repository.get_job(job_id)
+            return job is not None
+        except BSONError as e:  # Catch all mongoDB error
+            return False
+
+    return decorator
+
+
+@bp.before_request
+@token_required
+@validate_job_id
+def before_request():
+    """Validate user and job_id at all action endpoints."""
+    pass
```

Comment:
Validate all `job_id` prior to all requests to `server.actions` and standardize error handling.